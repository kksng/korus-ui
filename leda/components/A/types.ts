import React from 'react';
import { CustomEventHandler } from '../../commonTypes';

export interface AProps extends React.HTMLAttributes<HTMLAnchorElement> {
  /** Текст ссылки */
  children?: React.ReactNode,
  /** Ссылка */
  href?: string,
  /** Обработчик клика */
  onClick?: CustomEventHandler<React.MouseEvent<HTMLAnchorElement>>,
  /** Реф */
  ref?: React.Ref<ARefCurrent>,
  /** Если передано false, компонент не будет рендериться */
  shouldRender?: boolean,
  /** Классы переданные через _ */
  [x: string]: unknown,
}

export interface ARefCurrent {
  wrapper: HTMLAnchorElement | null,
}
