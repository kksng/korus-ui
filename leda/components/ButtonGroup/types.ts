import * as React from 'react';
import { PartialGlobalDefaultTheme } from '../../utils/useTheme';
import { COMPONENTS_NAMESPACES } from '../../constants';
import {
  ArrayElement, CustomEventHandler, CustomRender, SetState, SomeObject,
} from '../../commonTypes';
import { ButtonProps } from '../Button/types';
import { ValidationProps } from '../Validation/types';

export type Value = string | SomeObject | number | null;

export interface ResetEvent<T = Value | Value[]> {
  component: {
    value?: T, // Значение - элемент из data
    name?: string,
  },
}

export interface ButtonClickEvent<T = Value | Value[]> extends React.MouseEvent {
  component: {
    value: T, // Значение - элемент из data
    name?: string,
  },
}

export type ChangeEvent<T> = ButtonClickEvent<T> | ResetEvent<T>;

export interface ButtonGroupProps<T extends Value | Value[] = Value | Value[]> extends ValidationProps {
  /** Кастомизация кнопки при передачи data. По дефолту используется L.Button */
  buttonRender?: CustomRender<ButtonGroupProps, ButtonGroupState, ButtonProps>,
  /** Данные для элементов. Массив обьектов или строк или чисел. ВАЖНО! В компоненте не может быть двух кнопок с одинаковым текстом */
  data?: ArrayElement<T>[],
  /** Значение по-умолчанию */
  defaultValue?: Value | Value[],
  /** Выключенное состояние компонента */
  isDisabled?: boolean,
  /** Обработчик события изменения активного айтема. Отдает value и index */
  onChange?: (ev: ChangeEvent<T>) => void,
  /** Обработчик клика */
  onClick?: React.MouseEventHandler<HTMLElement>,
  /** При передаче массива обьектов указать текстовое поле из которого брать значение */
  textField?: string,
  /** Тип компонента, если radio - может выбрана только одна кнопка, иначе - несколько. По-умолчанию radio */
  type?: 'radio' | 'checkbox',
  /** Реф */
  ref?: React.Ref<ButtonGroupRefCurrent>,
  /** Значение активного элемента. Использовать при контролируемом режиме */
  value?: T,
  /** Тема компонента */
  theme?: PartialGlobalDefaultTheme[typeof COMPONENTS_NAMESPACES.buttonGroup],
  /** Кастомный рендер для враппера */
  wrapperRender?: CustomRender<ButtonGroupProps, ButtonGroupState, WrapperProps>,
  /** Классы переданные через _ */
  [x: string]: unknown,
}

export interface ButtonGroupState {
  value: Value | Value[],
}

export interface ButtonGroupRefCurrent {
  wrapper: HTMLElement | null,
}

export interface WrapperProps extends React.HTMLAttributes<HTMLElement> {
  ref?: React.Ref<ButtonGroupRefCurrent | null>,
}

export interface ChangeData {
  value: ButtonGroupProps['value'],
  setUncontrolledValue: SetState<ButtonGroupProps['value']>,
  validateCurrent: (value: ButtonGroupProps['value']) => boolean,
}
