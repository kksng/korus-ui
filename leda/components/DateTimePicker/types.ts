import * as React from 'react';
import {
  BlurEvent, ChangeEvent, FocusEvent, DateTimeInputProps,
  DateTimeInputRefCurrent, TimeLimits,
} from '../../src/DateTimeInput/types';

export interface DateTimePickerProps extends DateTimeInputProps {
  /** Формат отображаемой и вводимой даты. По-умолчанию "dd.MM.yyyy hh:mm" */
  format?: string,
  /** Признак принудительного открытия календаря */
  isOpen?: boolean,
  /** Признак отключения инпута */
  isDisabled?: boolean,
  /** Максимально доступная дата для выбора */
  max?: Date,
  /** Минимально доступная дата для выбора */
  min?: Date,
  /** Имя поля ввода */
  name?: string,
  /** Функция обратного вызова при потере фокуса */
  onBlur?: (ev: BlurEvent) => void,
  /** Функция обратного вызова при изменении значения в поле ввода */
  onChange?: (ev: ChangeEvent) => void,
  /**  Функция обратного вызова при нажатии Enter */
  onEnterPress?: (ev: ChangeEvent) => void,
  /** Функция обратного вызова при фокусе поля ввода */
  onFocus?: (ev: FocusEvent) => void,
  /** Плейсхолдер для поля ввода */
  placeholder?: string,
  /** Минимальное время */
  timeMin?: TimeLimits,
  /** Максимальное время */
  timeMax?: TimeLimits,
  /** Реф */
  ref?: React.Ref<DateTimeInputRefCurrent>,
  /** Значение поля ввода */
  value?: string | Date | null,
  /** Классы переданные через _ */
  [x: string]: unknown,
}
