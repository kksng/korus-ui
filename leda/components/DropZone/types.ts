import * as React from 'react';

import { CustomRender, CustomEventHandler } from '../../commonTypes';
import { PartialGlobalDefaultTheme } from '../../utils/useTheme';
import { COMPONENTS_NAMESPACES, FileErrorCodes } from '../../constants';
import { globalDefaultTheme } from '../LedaProvider';
import { ValidationProps } from '../Validation/types';

export { FileErrorCodes } from '../../constants';

/**
 * Properties of accepted files list
 */
export interface AcceptedFilesProps {
  /** Child React nodes */
  children?: React.ReactNode,
  /** Child DOM nodes */
  dropZoneFilesNode?: Element,
  /** onChange callback */
  onChange?: ChangeEventHandler,
}

/**
 * Change event
 */
export interface ChangeEvent {
  component: {
    /** Files that were added */
    dropped: DropZoneFiles,
    /** Actual state */
    value: DropZoneFiles,
    /** File that was removed */
    removedFile?: FileType,
  },
}

/**
 * Change event handler
 */
export type ChangeEventHandler = (
  /** Accepted files */
  accepted: FileType[],
  /** Rejected files */
  rejected: FileType[],
  /** Drag event or mouse event */
  ev?: React.DragEvent<HTMLElement> | React.MouseEvent<HTMLElement>,
  /** Removed file */
  removedFile?: FileType
) => void;

/**
 * Custom React elements of DropZone component
 */
export interface CustomElements {
  /** React component for accepted files list wrapper */
  AcceptedFiles: React.FC<AcceptedFilesProps>,
  /** React component for rejected files list wrapper */
  RejectedFiles: React.FC<RejectedFilesProps>,
  /** React component for info block */
  Info: React.FC<InfoProps>,
  /** React component for upload button */
  UploadButton: React.FC<UploadButtonProps>,
  /** React component for wrapper */
  Wrapper: React.FC<WrapperProps>,
}

/**
 * DropZone error object
 */
export interface DropZoneError {
  /** Error message */
  message: string,
  /** Error code */
  errorCode: FileErrorCodes,
}

/**
 * DropZone file type
 */
export interface DropZoneFileType extends File {
  /** File path */
  path?: string,
  /** File preview */
  preview?: string,
  /** Date and time when file was last modified */
  lastModified: number,
  /** Error code */
  errorCode?: FileErrorCodes,
}

/**
 * DropZone files
 */
export interface DropZoneFiles {
  /** List of accepted files */
  acceptedFiles: FileType[],
  /** List of rejected files by type or size */
  rejectedFiles: FileType[],
}

export interface DropZoneFilesProps {
  /** Массив файлов, который будет отображен в виде списка */
  files: FileType[],
  /** Обработчик удаления файла */
  onChange: (
    acceptedFiles: FileType[],
    rejectedFiles: FileType[],
    ev: React.DragEvent<HTMLDivElement> | React.MouseEvent<HTMLAnchorElement>,
    removedFile: FileType,
  ) => void,
  /** Флаг обозначающий нужно ли отображать файлы */
  shouldRender: boolean,
  /** Принятые и отклоненные файлы */
  value: DropZoneState | NonNullable<DropZoneProps['value']>,
  /** Тема компонента */
  theme: typeof globalDefaultTheme[typeof COMPONENTS_NAMESPACES.dropZone],
  /** Признак отключения дропзоны */
  isDisabled?: boolean,
}

export interface DropZoneProps extends ValidationProps {
  /** Отображение добавленных файлов */
  acceptedFilesRender?: CustomRender<DropZoneProps, DropZoneState, AcceptedFilesProps>,
  /** Разрешенные типы файлов, см. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input#Attributes. Передача нескольких типов файлов происходит через запятую (.png, image/jpeg). allowedFiles и forbiddenFiles вместе не могут находиться. */
  allowedFiles?: string,
  /** Классы, применяемые к компоненту */
  className?: string,
  /** DOM-нода в которой будет отрендерен список файлов */
  dropZoneFilesNode?: Element,
  /** Запрещенные типы файлов. см. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input#Attributes. Передача нескольких типов файлов происходит через запятую (.png, image/jpeg). allowedFiles и forbiddenFiles вместе не могут находиться. */
  forbiddenFiles?: string,
  /** Кастомизация описания компонента */
  infoRender?: CustomRender<DropZoneProps, DropZoneState, InfoProps>,
  /** Признак отключения дропзоны */
  isDisabled?: boolean,
  /** Максимальная длина имени файла, по-умолчанию 255 символов */
  maxFileNameLength?: number,
  /** Максимальное количество файлов */
  maxFilesNumber?: number,
  /** Максимальный размер файла, в байтах */
  maxFileSize?: number,
  /** Минимальный размер файла, в байтах */
  minFileSize?: number,
  /** Функция обратного вызова для метода onChange */
  onChange?: (event: ChangeEvent) => void,
  /** Функция обратного вызова для метода onClick */
  onClick?: (event: React.MouseEvent<HTMLDivElement>) => void,
  /** Отображение отклоненных файлов */
  rejectedFilesRender?: CustomRender<DropZoneProps, DropZoneState, RejectedFilesProps>,
  /** Реф */
  ref?: React.Ref<DropZoneRefCurrent>,
  /** Тема для компонента */
  theme?: PartialGlobalDefaultTheme[typeof COMPONENTS_NAMESPACES.dropZone],
  /** Текст кнопки загрузки файла, может принимать JSX */
  uploadButtonRender?: CustomRender<DropZoneProps, DropZoneState, UploadButtonProps>,
  /** Добавленные файлы */
  value?: DropZoneFiles | null,
  /** Кастомизация враппера */
  wrapperRender?: CustomRender<DropZoneProps, DropZoneState, WrapperProps>,
  /** Классы переданные через _ */
  [x: string]: unknown,
}

/**
 * DropZone current ref
 */
export interface DropZoneRefCurrent {
  /** Wrapper element */
  wrapper: HTMLElement | null,
  /** Input element */
  input: HTMLInputElement | null,
}

/**
 * DropZone state
 */
export type DropZoneState = DropZoneFiles;

/**
 * External file
 */
export interface ExternalFile {
  /** File name. Required for displaying in the list and deleting */
  name: string,
  /** Link to download the file. If available, the file will be displayed in the list as available for download */
  link?: string,
  /** Synonym for the file name. Required for displaying in the list and deleting */
  path?: string,
  /** File type */
  type?: string,
}

/**
 * Possible file types
 */
export type FileType = ExternalFile | DropZoneFileType;

/**
 * Properties of info block component
 */
export interface InfoProps {
  /** Child React nodes */
  children?: React.ReactNode,
  /** Custom class names */
  className?: string,
}

/**
 * Properties of rejected files list component
 */
export interface RejectedFilesListProps {
  /** Theme */
  theme: typeof globalDefaultTheme[typeof COMPONENTS_NAMESPACES.dropZone],
  /** Value got from state or props */
  value: DropZoneState | NonNullable<DropZoneProps['value']>,
  /** Maximum files number allowed */
  maxFilesNumber?: number,
}

/**
 * Properties of rejected files list wrapper component
 */
export interface RejectedFilesProps {
  /** Child React nodes */
  children?: React.ReactNode,
  /** Custom class names */
  className?: string,
}

/**
 * Properties of upload button component
 */
export interface UploadButtonProps {
  /** Child React nodes */
  children?: React.ReactNode,
  /** Custom class names */
  className?: string,
  /** onClick callback */
  onClick?: CustomEventHandler<React.MouseEvent>,
  /** Custom classes passed through _ */
  [x: string]: unknown,
}

/**
 * Properties of wrapper component
 */
export interface WrapperProps {
  /** Child React nodes */
  children?: React.ReactNode,
  /** Custom class names */
  className?: string,
  /** Component's ref */
  ref: React.Ref<DropZoneRefCurrent>,
}
