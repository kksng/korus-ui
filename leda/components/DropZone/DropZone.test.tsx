import React from 'react';
import { mount } from 'enzyme';
import {
  render, fireEvent, act, wait,
} from '@testing-library/react';
import toJson from 'enzyme-to-json';
import { DropZone } from './index';
import { FileErrorCodes } from './types';

const files = [
  new File(['file 1'], 'file_1.txt', { type: 'plain/text' }),
  new File(['file 2'], 'file_2.pdf', { type: 'application/pdf' }),
];

const externalFile = { name: 'file_3.txt', link: 'http://example.com/file.txt' };

const rejectedFile = new File(['rejected file'], 'rejected_file.txt', { type: 'plain/text' });
Object.assign(rejectedFile, { errorCode: FileErrorCodes.FileIsTooBig });

const mockObjectURL = 'blob://mockObjectURL';

beforeAll(() => {
  window.URL.createObjectURL = jest.fn(() => mockObjectURL);
});

describe('DropZone SNAPSHOTS', () => {
  it('should render', () => {
    const wrapper = mount(<DropZone onDrop={jest.fn()} onRemove={jest.fn()} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render file list', () => {
    const { container } = render(
      <DropZone
        value={{
          acceptedFiles: [
            ...files,
            externalFile,
          ],
          rejectedFiles: [rejectedFile],
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});

describe('DropZone HANDLERS', () => {
  it('should call onClick handler', () => {
    const handleClick = jest.fn();

    const { getByText } = render(<DropZone onClick={handleClick} />);

    fireEvent.click(getByText('Выбрать...'));

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('should call onChange handler on drop files into the dropZone', async () => {
    const handleChange = jest.fn();

    const eventMatcher = expect.objectContaining({
      component: expect.objectContaining({
        dropped: {
          acceptedFiles: [
            expect.objectContaining(files[0]),
            expect.objectContaining(files[1]),
          ],
          rejectedFiles: [],
        },
        value: {
          acceptedFiles: [
            expect.objectContaining(files[0]),
            expect.objectContaining(files[1]),
          ],
          rejectedFiles: [],
        },
      }),
    });

    const component = <DropZone onChange={handleChange} />;

    const { rerender, container, queryByText } = render(component);

    const dropZoneContent = container.querySelector('div.dropzone-content');

    expect(dropZoneContent).toBeInTheDocument();

    const data = {
      dataTransfer: {
        files,
        items: files.map((file) => ({
          kind: 'file',
          type: file.type,
          getAsFile: () => file,
        })),
        types: ['Files'],
      },
    };

    const event = new Event('drop', { bubbles: true });
    Object.assign(event, data);

    fireEvent(dropZoneContent as Element, event);

    await act(() => wait(() => rerender(component)));

    expect(queryByText('file_1.txt')).toBeInTheDocument();
    expect(queryByText('file_2.pdf')).toBeInTheDocument();

    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(handleChange).toHaveBeenLastCalledWith(eventMatcher);
  });

  /*
   TODO fix the bug with incorrect "dropped" value in the change event triggered by removing file
   Now: "dropped" equals "value" before remove
   Expected: "dropped" empty
   */
  // eslint-disable-next-line jest/no-disabled-tests
  it.skip('should call onChange handler on remove file', () => {
    const handleChange = jest.fn();

    const eventMatcher = expect.objectContaining({
      component: expect.objectContaining({
        dropped: {
          acceptedFiles: [],
          rejectedFiles: [],
        },
        value: {
          acceptedFiles: [],
          rejectedFiles: [],
        },
        removedFile: [externalFile],
      }),
    });

    const { container } = render(
      <DropZone
        onChange={handleChange}
        value={{
          acceptedFiles: [externalFile],
          rejectedFiles: [],
        }}
      />,
    );

    const removeFileIcon = container.querySelector('i.dropzone-delete-icon');

    expect(removeFileIcon).toBeInTheDocument();

    fireEvent.click(removeFileIcon as Element);

    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(handleChange).toHaveBeenLastCalledWith(eventMatcher);
  });
});

describe('DropZone ATTRIBUTES', () => {
  it('should have className, change classes through props and className should not change prop-classes', () => {
    const wrapper = mount(<DropZone _box onDrop={jest.fn()} onRemove={jest.fn()} />);

    expect(wrapper.find('Div').first().hasClass('box')).toBeTruthy();


    wrapper.setProps({ _active: true, _box: false });

    expect(wrapper.find('Div').first().hasClass('box')).toBeFalsy();

    expect(wrapper.find('Div').first().hasClass('active')).toBeTruthy();


    wrapper.setProps({ className: 'testClass' });

    expect(wrapper.find('Div').first().hasClass('box')).toBeFalsy();

    expect(wrapper.find('Div').first().hasClass('active')).toBeTruthy();

    expect(wrapper.find('Div').first().hasClass('testClass')).toBeTruthy();
  });

  it('should render files in custom HTMLElement', () => {
    const div = document.createElement('div');
    document.body.appendChild(div);
    const wrapper = mount(<DropZone dropZoneFilesNode={div} onDrop={jest.fn()} onRemove={jest.fn()} />);

    wrapper.setProps({ value: { acceptedFiles: files, rejectedFiles: [] } });

    expect(div.innerHTML.includes('file_1.txt')).toBeTruthy();
    expect(div.innerHTML.includes('file_2.pdf')).toBeTruthy();
  });

  it('should render descriptionText', () => {
    const wrapper = mount(<DropZone infoRender={() => <div className="dropzone-description">LETS DESCRIBE! Размер: до 100 Мб.</div>} onDrop={jest.fn()} onRemove={jest.fn()} />);

    expect(wrapper.find('div.dropzone-description').text()).toEqual('LETS DESCRIBE! Размер: до 100 Мб.');
  });

  it('should render width', () => {
    const wrapper = mount(<DropZone _width-20 onDrop={jest.fn()} onRemove={jest.fn()} />);

    expect(wrapper.find('div').first().getDOMNode().classList).toContain('width-20');
  });

  it('should customize upload button', () => {
    const wrapper = mount(
      <DropZone
        onDrop={jest.fn()}
        onRemove={jest.fn()}
        uploadButtonRender={() => <span className="test-class">Drop me</span>}
      />,
    );

    expect(wrapper.find('.test-class').text()).toEqual('Drop me');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should customize description text', () => {
    const wrapper = mount(
      <DropZone
        onDrop={jest.fn()}
        onRemove={jest.fn()}
        infoRender={() => <span className="dropzone-description">Drop here</span>}
      />,
    );

    expect(wrapper.find('span.dropzone-description').text()).toEqual('Drop here');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
