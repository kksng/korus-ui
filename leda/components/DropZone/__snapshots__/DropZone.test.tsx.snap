// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropZone ATTRIBUTES should customize description text 1`] = `
<DropZone
  infoRender={[Function]}
  onDrop={[MockFunction]}
  onRemove={[MockFunction]}
>
  <Wrapper
    className="dropzone-wrapper"
  >
    <Div
      className="dropzone-wrapper"
    >
      <div
        className="dropzone-wrapper"
      >
        <Div
          className="dropzone-content"
          onBlur={[Function]}
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <div
            className="dropzone-content"
            onBlur={[Function]}
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            <input
              autoComplete="off"
              multiple={true}
              onChange={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              tabIndex={-1}
              type="file"
            />
            <UploadButton
              className="dropzone-button"
            >
              <Button
                className="dropzone-button"
              >
                <button
                  className="button-wrapper dropzone-button"
                  onClick={[Function]}
                  type="button"
                >
                  Выбрать...
                </button>
              </Button>
            </UploadButton>
             
            <Info
              className="dropzone-description"
            >
              <span
                className="dropzone-description"
              >
                Drop here
              </span>
            </Info>
          </div>
        </Div>
      </div>
    </Div>
  </Wrapper>
  <InvalidMessageWrapper>
    <InvalidMessage
      isValid={true}
      messages={Array []}
    >
      <Component
        isValid={true}
        messages={Array []}
      />
    </InvalidMessage>
  </InvalidMessageWrapper>
  <RejectedFiles
    className="rejected-list margin-none"
  >
    <Ul
      className="rejected-list margin-none"
    >
      <ul
        className="rejected-list margin-none"
      >
        <Component
          theme={
            Object {
              "button": "dropzone-button",
              "content": "dropzone-content",
              "description": "dropzone-description",
              "disabled": "disabled",
              "fileDeleteIcon": "dropzone-delete-icon",
              "fileDeleteLink": "dropzone-file-delete internal",
              "fileDownloadLink": "dropzone-file-download internal",
              "invalid": "danger",
              "rejectedFilesWrapper": "rejected-list margin-none",
              "rejectedMessage": "dropzone-rejected-message txt-danger",
              "required": "required",
              "status": "dropzone-status",
              "wrapper": "dropzone-wrapper",
            }
          }
          value={
            Object {
              "acceptedFiles": Array [],
              "rejectedFiles": Array [],
            }
          }
        />
      </ul>
    </Ul>
  </RejectedFiles>
  <AcceptedFiles
    onChange={[Function]}
  >
    <AcceptedFiles
      onChange={[Function]}
    >
      <Component
        files={Array []}
        onChange={[Function]}
        shouldRender={false}
        theme={
          Object {
            "button": "dropzone-button",
            "content": "dropzone-content",
            "description": "dropzone-description",
            "disabled": "disabled",
            "fileDeleteIcon": "dropzone-delete-icon",
            "fileDeleteLink": "dropzone-file-delete internal",
            "fileDownloadLink": "dropzone-file-download internal",
            "invalid": "danger",
            "rejectedFilesWrapper": "rejected-list margin-none",
            "rejectedMessage": "dropzone-rejected-message txt-danger",
            "required": "required",
            "status": "dropzone-status",
            "wrapper": "dropzone-wrapper",
          }
        }
        value={
          Object {
            "acceptedFiles": Array [],
            "rejectedFiles": Array [],
          }
        }
      />
    </AcceptedFiles>
  </AcceptedFiles>
</DropZone>
`;

exports[`DropZone ATTRIBUTES should customize upload button 1`] = `
<DropZone
  onDrop={[MockFunction]}
  onRemove={[MockFunction]}
  uploadButtonRender={[Function]}
>
  <Wrapper
    className="dropzone-wrapper"
  >
    <Div
      className="dropzone-wrapper"
    >
      <div
        className="dropzone-wrapper"
      >
        <Div
          className="dropzone-content"
          onBlur={[Function]}
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <div
            className="dropzone-content"
            onBlur={[Function]}
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            <input
              autoComplete="off"
              multiple={true}
              onChange={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              tabIndex={-1}
              type="file"
            />
            <UploadButton
              className="dropzone-button"
            >
              <span
                className="test-class"
              >
                Drop me
              </span>
            </UploadButton>
             
            <Info
              className="dropzone-description"
            >
              <Span
                className="dropzone-description"
              >
                <span
                  className="dropzone-description"
                >
                  <Component>
                    <Div
                      _block-inline={true}
                    >
                      <div
                        className="block-inline"
                      >
                        Перетащите сюда файлы для загрузки.
                      </div>
                    </Div>
                  </Component>
                   
                  <Component>
                    <Div
                      _block-inline={true}
                    >
                      <div
                        className="block-inline"
                      >
                        Размер: до 104857600 байт.
                      </div>
                    </Div>
                  </Component>
                   
                  <Component />
                   
                  <Component />
                   
                  <Component />
                </span>
              </Span>
            </Info>
          </div>
        </Div>
      </div>
    </Div>
  </Wrapper>
  <InvalidMessageWrapper>
    <InvalidMessage
      isValid={true}
      messages={Array []}
    >
      <Component
        isValid={true}
        messages={Array []}
      />
    </InvalidMessage>
  </InvalidMessageWrapper>
  <RejectedFiles
    className="rejected-list margin-none"
  >
    <Ul
      className="rejected-list margin-none"
    >
      <ul
        className="rejected-list margin-none"
      >
        <Component
          theme={
            Object {
              "button": "dropzone-button",
              "content": "dropzone-content",
              "description": "dropzone-description",
              "disabled": "disabled",
              "fileDeleteIcon": "dropzone-delete-icon",
              "fileDeleteLink": "dropzone-file-delete internal",
              "fileDownloadLink": "dropzone-file-download internal",
              "invalid": "danger",
              "rejectedFilesWrapper": "rejected-list margin-none",
              "rejectedMessage": "dropzone-rejected-message txt-danger",
              "required": "required",
              "status": "dropzone-status",
              "wrapper": "dropzone-wrapper",
            }
          }
          value={
            Object {
              "acceptedFiles": Array [],
              "rejectedFiles": Array [],
            }
          }
        />
      </ul>
    </Ul>
  </RejectedFiles>
  <AcceptedFiles
    onChange={[Function]}
  >
    <AcceptedFiles
      onChange={[Function]}
    >
      <Component
        files={Array []}
        onChange={[Function]}
        shouldRender={false}
        theme={
          Object {
            "button": "dropzone-button",
            "content": "dropzone-content",
            "description": "dropzone-description",
            "disabled": "disabled",
            "fileDeleteIcon": "dropzone-delete-icon",
            "fileDeleteLink": "dropzone-file-delete internal",
            "fileDownloadLink": "dropzone-file-download internal",
            "invalid": "danger",
            "rejectedFilesWrapper": "rejected-list margin-none",
            "rejectedMessage": "dropzone-rejected-message txt-danger",
            "required": "required",
            "status": "dropzone-status",
            "wrapper": "dropzone-wrapper",
          }
        }
        value={
          Object {
            "acceptedFiles": Array [],
            "rejectedFiles": Array [],
          }
        }
      />
    </AcceptedFiles>
  </AcceptedFiles>
</DropZone>
`;

exports[`DropZone SNAPSHOTS should render 1`] = `
<DropZone
  onDrop={[MockFunction]}
  onRemove={[MockFunction]}
>
  <Wrapper
    className="dropzone-wrapper"
  >
    <Div
      className="dropzone-wrapper"
    >
      <div
        className="dropzone-wrapper"
      >
        <Div
          className="dropzone-content"
          onBlur={[Function]}
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <div
            className="dropzone-content"
            onBlur={[Function]}
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            <input
              autoComplete="off"
              multiple={true}
              onChange={[Function]}
              onClick={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              tabIndex={-1}
              type="file"
            />
            <UploadButton
              className="dropzone-button"
            >
              <Button
                className="dropzone-button"
              >
                <button
                  className="button-wrapper dropzone-button"
                  onClick={[Function]}
                  type="button"
                >
                  Выбрать...
                </button>
              </Button>
            </UploadButton>
             
            <Info
              className="dropzone-description"
            >
              <Span
                className="dropzone-description"
              >
                <span
                  className="dropzone-description"
                >
                  <Component>
                    <Div
                      _block-inline={true}
                    >
                      <div
                        className="block-inline"
                      >
                        Перетащите сюда файлы для загрузки.
                      </div>
                    </Div>
                  </Component>
                   
                  <Component>
                    <Div
                      _block-inline={true}
                    >
                      <div
                        className="block-inline"
                      >
                        Размер: до 104857600 байт.
                      </div>
                    </Div>
                  </Component>
                   
                  <Component />
                   
                  <Component />
                   
                  <Component />
                </span>
              </Span>
            </Info>
          </div>
        </Div>
      </div>
    </Div>
  </Wrapper>
  <InvalidMessageWrapper>
    <InvalidMessage
      isValid={true}
      messages={Array []}
    >
      <Component
        isValid={true}
        messages={Array []}
      />
    </InvalidMessage>
  </InvalidMessageWrapper>
  <RejectedFiles
    className="rejected-list margin-none"
  >
    <Ul
      className="rejected-list margin-none"
    >
      <ul
        className="rejected-list margin-none"
      >
        <Component
          theme={
            Object {
              "button": "dropzone-button",
              "content": "dropzone-content",
              "description": "dropzone-description",
              "disabled": "disabled",
              "fileDeleteIcon": "dropzone-delete-icon",
              "fileDeleteLink": "dropzone-file-delete internal",
              "fileDownloadLink": "dropzone-file-download internal",
              "invalid": "danger",
              "rejectedFilesWrapper": "rejected-list margin-none",
              "rejectedMessage": "dropzone-rejected-message txt-danger",
              "required": "required",
              "status": "dropzone-status",
              "wrapper": "dropzone-wrapper",
            }
          }
          value={
            Object {
              "acceptedFiles": Array [],
              "rejectedFiles": Array [],
            }
          }
        />
      </ul>
    </Ul>
  </RejectedFiles>
  <AcceptedFiles
    onChange={[Function]}
  >
    <AcceptedFiles
      onChange={[Function]}
    >
      <Component
        files={Array []}
        onChange={[Function]}
        shouldRender={false}
        theme={
          Object {
            "button": "dropzone-button",
            "content": "dropzone-content",
            "description": "dropzone-description",
            "disabled": "disabled",
            "fileDeleteIcon": "dropzone-delete-icon",
            "fileDeleteLink": "dropzone-file-delete internal",
            "fileDownloadLink": "dropzone-file-download internal",
            "invalid": "danger",
            "rejectedFilesWrapper": "rejected-list margin-none",
            "rejectedMessage": "dropzone-rejected-message txt-danger",
            "required": "required",
            "status": "dropzone-status",
            "wrapper": "dropzone-wrapper",
          }
        }
        value={
          Object {
            "acceptedFiles": Array [],
            "rejectedFiles": Array [],
          }
        }
      />
    </AcceptedFiles>
  </AcceptedFiles>
</DropZone>
`;

exports[`DropZone SNAPSHOTS should render file list 1`] = `
<div>
  <div
    class="dropzone-wrapper"
  >
    <div
      class="dropzone-content"
      tabindex="0"
    >
      <input
        autocomplete="off"
        multiple=""
        style="display: none;"
        tabindex="-1"
        type="file"
      />
      <button
        class="button-wrapper dropzone-button"
        type="button"
      >
        Выбрать...
      </button>
       
      <span
        class="dropzone-description"
      >
        <div
          class="block-inline"
        >
          Перетащите сюда файлы для загрузки.
        </div>
         
        <div
          class="block-inline"
        >
          Размер: до 104857600 байт.
        </div>
         
        
         
        
         
        
      </span>
    </div>
  </div>
  <ul
    class="rejected-list margin-none"
  >
    <li>
      <span
        class="dropzone-rejected-message txt-danger"
      >
        'rejected_file.txt': - Превышен максимальный размер файла
      </span>
    </li>
  </ul>
  <ul>
    <li>
      <div
        style="display: none;"
      />
      <a
        class="pointer"
      >
        <i
          class="dropzone-delete-icon"
        />
      </a>
      <div
        style="display: none;"
      />
      <a
        download="file_1.txt"
        href="blob://mockObjectURL"
        theme="dropzone-file-download internal"
      >
        file_1.txt
      </a>
    </li>
    <li>
      <div
        style="display: none;"
      />
      <a
        class="pointer"
      >
        <i
          class="dropzone-delete-icon"
        />
      </a>
      <div
        style="display: none;"
      />
      <a
        download="file_2.pdf"
        href="blob://mockObjectURL"
        theme="dropzone-file-download internal"
      >
        file_2.pdf
      </a>
    </li>
    <li>
      <div
        style="display: none;"
      />
      <a
        class="pointer"
      >
        <i
          class="dropzone-delete-icon"
        />
      </a>
      <div
        style="display: none;"
      />
      <a
        download="file_3.txt"
        href="http://example.com/file.txt"
        theme="dropzone-file-download internal"
      >
        file_3.txt
      </a>
    </li>
  </ul>
</div>
`;
